File: py/p01.py

from utils import data19

def fuel(mass):
    return max(0, mass // 3 - 2)

modules = [int(x) for x in data19(1).splitlines()]


fuels = [fuel(mass) for mass in modules]

print(f"Answer1: {sum(fuels)}")

total = sum(fuels)

while sum(fuels) > 0:
    fuels = [fuel(mass) for mass in fuels]
    total += sum(fuels)

print(f"Answer2: {total}")

File: 2019/nim/p01.nim
import seqUtils, strUtils, math

func fuel(mass: seq[int]): seq[int] = mass.mapIt(max(0, it div 3 - 2))

let mass = toSeq(lines("../input/01.txt")).map(parseInt).fuel

when isMainModule:
  echo "Answer1: ", mass.sum

  var numbers = mass
  var total = numbers.sum

  while numbers.sum > 0:
    numbers = numbers.fuel
    total += numbers.sum

  echo "Answer2: ", total

